src/store/chat/saga.test.ts

```diff
@@ -140,11 +178,47 @@ describe(closeErrorDialog, () => {
     expect(storeState.chat.isConversationErrorDialogOpen).toBe(false);
   });
 
+  it('sets the join room error content state', async () => {
+    const initialState = new StoreBuilder().withChat({
+      joinRoomErrorContent: {
+        header: 'Access Denied',
+        body: 'You do not have permission to join this conversation.',
+      },
+    });
+    const { storeState } = await subject(closeErrorDialog).withReducer(rootReducer, initialState.build()).run();
+
+    expect(storeState.chat.joinRoomErrorContent).toBe(null);
+  });
+
   it('opens the first conversation', async () => {
     await subject(closeErrorDialog).withReducer(rootReducer).call(openFirstConversation).run();
   });
 });
 
+describe(joinRoom, () => {
+  describe('error scenarios', () => {
+    const roomIdOrAlias = 'some-room-id-or-alias';
+    it.each([
+      [JoinRoomApiErrorCode.ROOM_NOT_FOUND, ERROR_DIALOG_CONTENT[JoinRoomApiErrorCode.ROOM_NOT_FOUND]],
+      [JoinRoomApiErrorCode.ACCESS_TOKEN_REQUIRED, ERROR_DIALOG_CONTENT[JoinRoomApiErrorCode.ACCESS_TOKEN_REQUIRED]],
+      [JoinRoomApiErrorCode.GENERAL_ERROR, ERROR_DIALOG_CONTENT[JoinRoomApiErrorCode.GENERAL_ERROR]],
+    ])('handles error "%s"', async (errorCode, expectedErrorContent) => {
```

Comment:
This seems a little complex, could you please explain this?

Reply:
`it.each` is used to simplify writing multiple, similar tests. I've now changed this and added an individual test for each error case instead so its a little easier to follow. 

` const expectedErrorContent = ERROR_DIALOG_CONTENT[JoinRoomApiErrorCode.ROOM_NOT_FOUND];` 

is:

```
 [JoinRoomApiErrorCode.ROOM_NOT_FOUND]: {
    header: 'Conversation Not Found',
    body: 'This conversation does not exist.',
  }
```

Hopefully the changes will help make more sense!

Reply:
thanks